cmake/Modules/FindComputeCpp.cmake

```diff
@@ -276,39 +370,49 @@ function(__build_spir targetName sourceFile binaryDir fileCounter)
             -isystem ${COMPUTECPP_INCLUDE_DIRECTORY}
             ${device_compiler_includes}
             -o ${outputSyclFile}
-            -c ${sourceFile}
-    DEPENDS ${sourceFile}
-    IMPLICIT_DEPENDS CXX ${sourceFile}
-    WORKING_DIRECTORY ${binaryDir}
+            -c ${SDK_BUILD_IR_SOURCE}
+    DEPENDS ${ir_dependencies}
+    IMPLICIT_DEPENDS CXX ${SDK_BUILD_IR_SOURCE}
+    WORKING_DIRECTORY ${SDK_BUILD_IR_BINARY_DIR}
     COMMENT "Building ComputeCpp integration header file ${outputSyclFile}")
 
-  # Name:
-  # (user-defined name)_(source file)_(counter)_ih
+  # Name: (user-defined name)_(source file)_(counter)_ih
   set(headerTargetName
-    ${targetName}_${sourceFileName}_${fileCounter}_ih)
+    ${SDK_BUILD_IR_TARGET}_${sourceFileName}_${SDK_BUILD_IR_COUNTER}_ih)
 
   if(NOT MSVC)
     # Add a custom target for the generated integration header
     add_custom_target(${headerTargetName} DEPENDS ${outputSyclFile})
-
-    # Add a dependency on the integration header
-    add_dependencies(${targetName} ${headerTargetName})
+    add_dependencies(${SDK_BUILD_IR_TARGET} ${headerTargetName})
   endif()
 
   # This property can be set on a per-target basis to indicate that the
   # integration header should appear after the main source listing
-  get_property(includeAfter TARGET ${targetName}
+  get_property(includeAfter TARGET ${SDK_BUILD_IR_TARGET}
       PROPERTY COMPUTECPP_INCLUDE_AFTER)
 
   if(includeAfter)
     # Change the source file to the integration header - e.g.
     # g++ -c source_file_name.cpp.sycl
-    set_property(TARGET ${targetName} PROPERTY SOURCES ${outputSyclFile})
+    get_target_property(current_sources ${SDK_BUILD_IR_TARGET} SOURCES)
+    # Remove absolute path to source file
+    list(REMOVE_ITEM current_sources ${SDK_BUILD_IR_SOURCE})
+    # Remove relative path to source file
+    string(REPLACE "${CMAKE_CURRENT_SOURCE_DIR}/" ""
+      rel_source_file ${SDK_BUILD_IR_SOURCE}
+    )
+    list(REMOVE_ITEM current_sources ${rel_source_file})
+    # Add SYCL header to source list
+    list(APPEND current_sources ${outputSyclFile})
+    set_property(TARGET ${SDK_BUILD_IR_TARGET}
+      PROPERTY SOURCES ${current_sources})
     # CMake/gcc don't know what language a .sycl file is, so tell them
     set_property(SOURCE ${outputSyclFile} PROPERTY LANGUAGE CXX)
```

Comment:
It should also have the [HEADER_FILE_ONLY](https://cmake.org/cmake/help/v3.12/prop_sf/HEADER_FILE_ONLY.html) property.

Reply:
Done! But only in the "non-include-after" case.